src/layers/learning/channelwise_scale_bias.cpp

```diff
@@ -0,0 +1,136 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "lbann/layers/learning/channelwise_scale_bias.hpp"
+
+namespace lbann {
+
+template <>
+void channelwise_scale_bias_layer<data_layout::DATA_PARALLEL,El::Device::CPU>
+     ::fp_compute() {
+
+  // Local matrices
+  const auto& local_input = dynamic_cast<const CPUMat&>(get_local_prev_activations());
+  auto& local_output = dynamic_cast<CPUMat&>(get_local_activations());
+  const auto& local_weights = dynamic_cast<const CPUMat&>(m_weights[0]->get_values().LockedMatrix());
+  const auto local_scale = El::LockedView(local_weights,
+                                          El::ALL, El::IR(0));
+  const auto local_bias = El::LockedView(local_weights,
+                                         El::ALL, El::IR(1));
+
+  // Dimensions
+  // Note: channel_size is the number of input entries per channel and
+  // local_width is the number of local mini-batch samples.
```

Comment:
Maybe we should just name the variable `local_mini_batch_size` then?

Reply:
I don't think there's much difference. For me this does a better job of conveying the intuition that we're iterating through the rows and columns of a matrix.